server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -139,6 +149,41 @@ public String getApplicationName() {
     return applicationName;
   }
 
+  /**
+   * Get the ServiceLock for this server process. May return null if called before the lock is
+   * acquired.
+   *
+   * @return lock ServiceLock or null
+   */
+  public abstract ServiceLock getLock();
```

Comment:
I think you could avoid making this abstract and needing to implement it in the child classes if each child class called the super constructor and provided a `Supplier<ServiceLock>`, then this impl would just be:

```suggestion
  public ServiceLock getLock() {
    serviceLockSupplier.get();
  }
```

That might also avoid issues with handling null here, because the supplier would likely be blocking.

Reply:
But wouldn't that require an AtomicReference<ServiceLock> or something so that the Supplier can just call AtomicReference.get() and the server call AtomicReference.set(). I'm not sure that's any cleaner.